src/gui/DatabaseWidget.cpp

```diff
@@ -1061,6 +1075,37 @@ QStringList DatabaseWidget::customEntryAttributes() const
     return entry->attributes()->customKeys();
 }
 
+/*
+ * Restores the focus on the group and entry that was focused
+ * before the database was locked or reloaded.
+ */
+void DatabaseWidget::restoreGroupEntryFocus(Uuid groupUuid, Uuid entryUuid)
+{
+    Group* restoredGroup;
+    const QList<Group*> groups = m_db->rootGroup()->groupsRecursive(true);
+    for (Group* group : groups) {
+        if (group->uuid() == groupUuid) {
+            restoredGroup = group;
+            break;
+        }
+    }
+
+    if (!restoredGroup) {
```

Comment:
preferably this should be ```if (restoredGroup != nullptr)``` followed by the remainder of the function instead of an "early out" negative test for cleaner code.